lightning-invoice/src/payment.rs

```diff
@@ -773,25 +740,48 @@ where
 	}
 }
 
+/// Data structure used to track information about inflight HTLCs.
+pub struct InFlightHtlcs {
+	/// A map with liquidity value (in msat) keyed by a short channel id and the direction the HTLC
+	/// is traveling in. The direction boolean is determined by checking if the HTLC source's public
+	/// key is less than its destination. See [`InflightHtlcs::used_liquidity_msat`] for more
+	/// details.
+	pub map: HashMap<(u64, bool), u64>
+}
+
+impl InFlightHtlcs {
+	/// Creates a new instance with an optional map of liquidity information.
+	pub fn new(map: Option<HashMap<(u64, bool), u64>>) -> Self {
+		InFlightHtlcs { map: map.unwrap_or(HashMap::new()) }
+	}
+
+	/// Returns liquidity in msat given the public key of the HTLC source, destination, and short
+	/// channel id.
+	pub fn used_liquidity_msat(&self, source: &NodeId, destination: &NodeId, channel_scid: u64) -> Option<&u64> {
+		self.map.get(&(channel_scid, source < destination))
+	}
+}
+
 #[cfg(test)]
 mod tests {
 	use super::*;
 	use crate::{InvoiceBuilder, Currency};
-	use utils::create_invoice_from_channelmanager_and_duration_since_epoch;
+	use utils::{ScorerAccountingForInFlightHtlcs, create_invoice_from_channelmanager_and_duration_since_epoch};
 	use bitcoin_hashes::sha256::Hash as Sha256;
 	use lightning::ln::PaymentPreimage;
 	use lightning::ln::features::{ChannelFeatures, NodeFeatures, InitFeatures};
 	use lightning::ln::functional_test_utils::*;
 	use lightning::ln::msgs::{ChannelMessageHandler, ErrorAction, LightningError};
 	use lightning::routing::gossip::{EffectiveCapacity, NodeId};
 	use lightning::routing::router::{PaymentParameters, Route, RouteHop};
-	use lightning::routing::scoring::ChannelUsage;
+	use lightning::routing::scoring::{ChannelUsage, Score, LockableScore};
```

Comment:
Lexicographical order.

Reply:
Fixed in c5206b3bf646b61db957ef7dd80b79538114181d